MSAL/src/native_auth/public/MSALNativeAuthUserAccountResult+Internal.swift

```diff
@@ -26,25 +26,91 @@ import Foundation
+        params.correlationId = correlationId
+
+        let challengeTypes = MSALNativeAuthPublicClientApplication.convertChallengeTypes(configuration.challengeTypes)
+        let authority = try? MSALCIAMAuthority(url: configuration.authority.url)
+        let config = MSALPublicClientApplicationConfig(clientId: configuration.clientId,
+                                                       redirectUri: configuration.redirectUri,
+                                                       authority: authority)
+        config.bypassRedirectURIValidation = configuration.redirectUri == nil
+
+        guard let client = try? MSALNativeAuthPublicClientApplication(configuration: config, challengeTypes: challengeTypes)
+        else {
+            MSALLogger.log(
+                            level: .error,
+                            context: context,
+                            format: "Config or challenge types unexpectedly found nil."
+                        )
+            Task { await delegate.onAccessTokenRetrieveError(
+                error: RetrieveAccessTokenError(type: .generalError,
+                                                correlationId: correlationId ?? context.correlationId())) }
+            return
+        }
+
+        client.acquireTokenSilent(with: params) { [weak self] result, error in
+            guard let self = self else { return }
+
+            if let error = error as? NSError {
+                let accessTokenError = self.createRetrieveAccessTokenError(error: error, context: context)
+                Task { await delegate.onAccessTokenRetrieveError(error: accessTokenError) }
+                return
+            }
+
+            if let result = result {
+                let delegateDispatcher = CredentialsDelegateDispatcher(delegate: delegate, telemetryUpdate: nil)
+                let accessTokenResult = MSALNativeAuthTokenResult(accessToken: result.accessToken,
+                                                                  scopes: result.scopes,
+                                                                  expiresOn: result.expiresOn)
+                Task { await delegateDispatcher.dispatchAccessTokenRetrieveCompleted(result: accessTokenResult, correlationId: result.correlationId) }
+                return
+            }
+
+            Task { await delegate.onAccessTokenRetrieveError(error: RetrieveAccessTokenError(type: .generalError,
+                                                                                             correlationId: correlationId ?? context.correlationId())) }
         }
     }
+    
+    func createRetrieveAccessTokenError(error: NSError, context: MSALNativeAuthRequestContext) -> RetrieveAccessTokenError {
+        if let innerError = error.userInfo[NSUnderlyingErrorKey] as? NSError,
+           let message = innerError.userInfo[MSALErrorDescriptionKey] as? String,
+           let code = codeAndCorrelationIdFromMSALError(error: innerError).code,
```

Comment:
if you split the `codeAndCorrelationIdFromMSALError` in two functions you can avoid retrieving twice the code and correlationId at line 86 and 85.
So, this would be 
```suggestion
           let code = codeFromMSALError(error: innerError),
           let correlationId = correlationIdFromMSALError(error: innerError)
```

Reply:
Good idea. Done.